src/errors/error_handler.js

```diff
@@ -28,39 +50,47 @@ export default class ErrorHandler {
       errors,
     };
 
-    if (Array.isArray(errorParams.message)) {
-      errorParams.message = errorParams.message.join(', ');
+    try {
+      const messages = [];
+      this.traverseJsonElement(errorParams.message, messages);
+      errorParams.message = messages.join(', ');
+    } catch (e) {
+      const errorParams = {
+        message: Constants.ERROR_DESERIALIZATION,
+        code: 'ERROR_DESERIALIZATION_ERROR',
+      };
+      throw new EasyPostError(errorParams);
     }
 
     if (statusCode >= 300 && statusCode < 400) {
-      return new RedirectError(errorParams);
+      throw new RedirectError(errorParams);
```

Comment:
This function should not be throwing errors, but rather returning new instances (which are thrown by upstream functions that call this function)

Reply:
That's true. I changed this because the `handleApiError` function isn't throwing an error and can't really test it. I rolled this back and add `throw` in the unit tests. All good.